examples/common/eventdisplay/FairEveMCTracksEditor.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * FairEveMCTracksEditor.h
+ *
+ *  Created on: 27 sie 2020
+ *      Author: Daniel Wielanek
+ *		E-mail: daniel.wielanek@gmail.com
+ *		Warsaw University of Technology, Faculty of Physics
+ */
+#ifndef FAIREVEMCTRACKSEDITOR_H_
+#define FAIREVEMCTRACKSEDITOR_H_
+
+#include "FairEveCut.h"
+#include "FairEveMCTracks.h"
+
+#include <TGedFrame.h>
+
+class FairEveMCTracksEditor : public TGedFrame
+{
+    FairEveMinMaxCut *fPtCut;
+    FairEveMinMaxCut *fEtaCut;
+    FairEveMinMaxCut *fEnergyCut;
+    FairEveBoolCut *fPrimary;
+    FairEveBoolCut *fSecondary;
+    FairEveIntCut *fPdgCut;
+    FairEveMCTracks *fMCTracks;
```

Comment:
From the looks most of these members are leaking, are they not?

Reply:
I make them unique pointers, so I hope this solve the issue.
Btw. is the order of declaration access specifiers (public/protected/private) must be keep?
I noticed that I always start with private and ends with public, should I change order?
And one more question what an order I should use at the same level of access - members and later methods?

Reply:
> Btw. is the order of declaration access specifiers (public/protected/private) must be keep?
> I noticed that I always start with private and ends with public, should I change order?
> And one more question Btw. is the order of declaration access specifiers (public/protected/private) must be keep?

The order of class members determines the order of initialization (and potentially destruction in reverse-order of initialization) regardless of any access specifier. The order also determines at which offset in the type this member is stored in memory. However, the order of the access specifiers does not matter, it is just convenient to group many members with the same access specifier to not have to write it again many times - but you can have an access specifier as often as you want and also multiple times.

Reply:
Please add `#include <memory>` where you use [`std::unique_ptr<>`](https://en.cppreference.com/w/cpp/memory/unique_ptr), thx!

Reply:
One more question.
I disscused those updated with Florian and Muhammad, we decided that we have to break backward compatibility - I mean FairMCTracks will not be longer added to event display as a task.
I applied patched to proper macros but I cannot test some of them - e.g. examples/simulation/rutherford/macros/eventDisplay.C does't run because the function (eventdisplay) doesnt have any type to return.

Reply:
> but I cannot test some of them - e.g. examples/simulation/rutherford/macros/eventDisplay.C does't run because the function (eventdisplay) doesnt have any type to return.

This is a bug, thx for noticing! Feel free to apply the necessary fix, e.g.

```diff
--- a/examples/simulation/rutherford/macros/eventDisplay.C
+++ b/examples/simulation/rutherford/macros/eventDisplay.C
@@ -5,7 +5,7 @@
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
  *                  copied verbatim in the file "LICENSE"                       *
  ********************************************************************************/
-eventDisplay(TString mcEngine = "TGeant3")
+void eventDisplay(TString mcEngine = "TGeant3")
 {

     TString inFile = "data/test_" + mcEngine + ".mc.root";
```

Reply:
I think I fixed all of them.